api/krusty/gkesagenerator_test.go

```diff
@@ -0,0 +1,147 @@
+package krusty_test
+
+import (
+	"testing"
+
+	kusttest_test "sigs.k8s.io/kustomize/api/testutils/kusttest"
+)
+
+func TestGkeGenerator(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarness(t)
+	defer th.Reset()
+
+	th.WriteK(".", `
+resources:
+- resource.yaml
+
+gkeSaGenerator:
```

Comment:
I'd say put all the effort into the filter test for now.  

and/or drive this via the `generators` field e.g.
 - chartinflatorplugin_test.go
 - inlinetransformer_test.go

Reply:
Given that this is cloud-provider-specific, does it need to be a top-level field, or could we make it generators-only? Is there a need for a similar generator for other cloud providers? Are the inputs required similar enough to share an abstraction? If not will we accept a bunch of top-level fields?

Reply:
@KnVerey had a chat with @monopole this morning. We are thinking having one top level field that has a field like `cloud: gke` to tell us which cloud provider to generate for. When we want to have something for other cloud providers, we can add more options to this generator.